### DISCUSSION TITLE
Retrieve First Media Collection Using Spatie Package

### DISCUSSION THREAD

**Discussion by ousid** [filamentphp/filament#12029] [Help] [answered]
### Package

Table builder

### Package Version

v^3.2

### How can we help you?

I'm using `flament/tables` as standalone package, with custom liveiwre components, and I came across a situation, to show an image on the table (thumbnail) and I'm using the media library package, that filament provide.

### Steps to produce
- require `filament/tables`
- Implement it inside livewire component
- build-up the table columns
- Show the result in the browse

### The Problem
The problem that when I use `SpatieMediaLibraryImageColumn`, there's no API, or a method to retrieve the first media from the collection.

I have tried using `getImageUrl`, and I faced an error says:
`
Filament\Tables\Columns\Column::hasRelationship(): Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given
`

### Code Sample:

```php
// ...
public function table(Table $table): Table
{
  return $table
     ->query(
        Product::query()->with(['category', 'model', 'media'])
     )
     ->columns([
        Tables\Columns\SpatieMediaLibraryImageColumn::make('image')
        ->collection('images')
        ->getImageUrl(),
        // ...
     ])
}
```

Any help, or a solution, would be much appreciate it.

### Updates

I have came up with a temp solution, not sure if it's the correct solution, but it does the trick.

Simply use `defaultImageUrl` method, and pass the first media url

```php
// ...
->defaultImageUrl(
  fn (Model $record): ?string => $record->getFirstMediaUrl('your-collection')
)
``

**Comment by leandrocfe** [comment, marked as answer]
What about `Tables\Columns\SpatieMediaLibraryImageColumn::make('image')->limit(1)` ?

**Reply by ousid** [reply]
That works too! Thanks, @leandrocfe!

It's way cleaner, and efficient than the default column.

**Reply by abdulmannans-joistic** [reply]
How to retrive the last media or the latest one?
